openslides_backend/action/actions/mediafile/permission_mixin.py

```diff
@@ -0,0 +1,139 @@
+from typing import Any, Dict, List, Optional, Tuple
+
+from ....permissions.management_levels import OrganizationManagementLevel
+from ....permissions.permission_helper import has_organization_management_level
+from ....services.datastore.commands import GetManyRequest
+from ....shared.exceptions import (
+    ActionException,
+    MissingPermission,
+    PermissionException,
+)
+from ....shared.filters import And, Filter, FilterOperator, Not
+from ....shared.patterns import KEYSEPARATOR, Collection, FullQualifiedId
+from ...action import Action
+
+
+class MediafilePermissionMixin(Action):
+    """
+    Mixin to handle the check_permissions of mediafile actions.
+    Overwrite check_permissions() and get_meeting_id().
+    """
+
+    def check_permissions(self, instance: Dict[str, Any]) -> None:
+        collection, id_ = self.get_owner_data(instance)
+
+        self.check_parent_is_dir_and_owner(
+            instance.get("parent_id"), str(instance.get("owner_id"))
+        )
+        parent_id = instance.get("parent_id")
+        if not parent_id and instance.get("id"):
+            mediafile = self.datastore.get(
+                FullQualifiedId(self.model.collection, instance["id"]), ["parent_id"]
+            )
+            parent_id = mediafile.get("parent_id")
+        self.check_title_parent_unique(
+            instance.get("title"), parent_id, instance.get("id")
+        )
+
+        # handle organization permissions
+        if collection == "organization":
+            self.assert_not_anonymous()
+            self.check_token_unique(instance.get("token"), instance.get("id"))
+            if "access_group_ids" in instance:
+                raise PermissionException(
```

Comment:
This should be an `ActionException`, since the wrong payload was given. `PermissionError`s are for situations where the user has insufficent permissions to execute the given permissions, but otherwise the request is correct.

Reply:
Changed.